packages/generator/src/sdk-metadata/sdk-metadata-types.ts

```diff
@@ -0,0 +1,42 @@
+export interface SdkMetadataHeader {
+  name: string; // API_BANKDETAIL_SRV
+  type: 'odata' | 'rest' | 'soap';
+  version: string;
+}
+
+// If generationAndUsage is not possible  e.g. SOAP  service for JS we do not put it in the array
+export interface Client {
+  language: 'java' | 'javascript';
+  pregeneratedLibrary: PregeneratedLibrary | undefined; // undefined if no pregenerated exists
+  generationAndUsage: GenerationAndUsage;
+}
+
+export interface PregeneratedLibrary {
+  version: string; // version of the client  library e.g. "1.23.0" - NOT the generator and NOT the API version
+  generatedAt: DateTimeString;
+  description: string; // "some-description-about-the-library",
+  installLibrarySnippet: string; // "npm -i @sap/cloud-sdk-vdm-bank-detail-service" or  "<dependencies><br /><dependency><br /><groupId>com.sap.cloud.sdk.s4hana</groupId><br /><artifactId>s4hana-api-odata</artifactId><br /><version>use-latest-version-here</version><br /></dependency><br /></dependencies><br />",
+  compatibilityNotes: '';// 'You have a minor version mismatch',...
+  repository: 'maven' | 'npm';
+  repositoryLink: UrlString; // "https://www.npmjs.com/package/@sap/cloud-sdk-vdm-bank-detail-service" or https://mvnrepository.com/artifact/com.sap.cloud.sdk.cloudplatform/cloudplatform-core
+  dependencyName: string; // "@sap/cloud-sdk-vdm-bank-detail-service" or com.sap.cloud.sdk.cloudplatform/cloudplatform-core
+}
+
+export interface Links {
+  sdkDocumentation: UrlString;
+  featureDocumentation: UrlString;
+  support: UrlString;
+  apiHubTutorial: UrlString;
+  generationManual: UrlString;
+}
+
+export interface GenerationAndUsage {
+  successfulGenerationVerified: boolean; // we have tested the API spec with the generator -> true, false if not tested
+  generationSteps: string; // "some-step<br />another step <br />another step"
+  apiSpecificUsage: string;
+  genericUsage: string; // import { <ServiceName> } from @sap/cloud-sdk-vdm-<service-name>-service<br /><ServiceName>.requestBuilder()<br />.<operationName>()<br />.execute(destination);
```

Comment:
I see lots of types are using `string`, but typically there are line breaks `<br />` used.
Could we use `string[]` and `join` later on?

Reply:
This type is meant a bit as the schema for the final web service. It is not optimal to use TS for that but it was so convinient because you can refactor a lot. I will create a new dummy type like `MultiLineText` to indicate that the field contains a text with multiple lines.